Lib/pathlib.py

```diff
@@ -1184,6 +1184,25 @@ def __new__(cls, *args, **kwargs):
             cls = WindowsPath if os.name == 'nt' else PosixPath
         return object.__new__(cls)
 
+    @classmethod
+    def from_uri(cls, uri):
+        """Return a new path from the given 'file' URI."""
+        uri = uri.removeprefix('file:')
```

Comment:
Should the method fail if the URI starts with something other than `file:`?

(looking at https://github.com/pypa/pip/blob/6f3a7181b6bfa0e95d479b3269baa0e551ff2cb2/src/pip/_internal/utils/urls.py#L26-L62)

Reply:
Quite possibly.

The main `PurePath()` / `Path()` constructor does not perform any validation (for example, it doesn't reject paths with null characters). I figured we could make `from_uri()` do the same, but it's a pretty weak rationale.